addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,434 @@
+		return false, err
+	}
+	pkgShortname = strings.Split(pkgRefName, ".")[0]
+
+	// package install name for core/additional packages in the workload cluster follows the <cluster name>-<addon short name> naming convention
+	// for management cluster, the naming convention for core/additional package installs is just <addon short name>
+	if clusterRole == clusterRoleWorkload {
+		pkgiName = util.GeneratePackageInstallName(clusterName, pkgRefName)
+	} else if clusterRole == clusterRoleManagement {
+		pkgiName = pkgShortname
+	}
+
+	pkgi := &kapppkgiv1alpha1.PackageInstall{}
+	objectKey := client.ObjectKey{Namespace: pkgiNamespace, Name: pkgiName}
+
+	// Note: use of the client (remote or not) depends on whether the corresponding cluster is
+	// workload or management which is determined in the Reconcile function
+	if err := c.Get(ctx, objectKey, pkgi); err != nil {
+		log.Error(err, fmt.Sprintf("unable to get PackageInstall '%s/%s'", pkgiNamespace, pkgiName))
+		return false, err
+	}
+
+	// For each package, create a single summary condition from the condition slice
+	pkgiCondition := util.SummarizeAppConditions(pkgi.Status.Conditions)
+
+	// Note: in case of encountering an unknown PackageInstall condition, just return err=nil and proceed with handling the next package
+	if pkgiCondition.Type == util.UnknownCondition {
+		log.Error(fmt.Errorf("unknown condition type for '%s/%s'", pkgi.Name, pkgi.Name), "unknown condition")
+		return false, nil
+	}
+
+	condition := runtanzuv1alpha3.Condition{
+		Type: runtanzuv1alpha3.ConditionType(strings.Title(pkgShortname)) + "-" +
+			runtanzuv1alpha3.ConditionType(pkgiCondition.Type),
+		Status:             pkgiCondition.Status,
+		UsefulErrorMessage: util.GetKappUsefulErrorMessage(pkgi.Status.UsefulErrorMessage),
+		LastTransitionTime: metav1.NewTime(time.Now().UTC().Truncate(time.Second)),
+	}
+
+	// Only add a new condition entry for the PackageInstall in the clusterBootstrapStatus in case it doesn't already exist
+	// If it does, just update it with the new condition
+	var conditionExists bool
+	var conditionChanged bool
+	for i, existingCond := range clusterBootstrapStatus.Conditions {
+		if !strings.Contains(string(existingCond.Type), strings.Title(pkgShortname)) {
+			continue
+		}
+		conditionExists = true
+		if !util.HasSameState(&clusterBootstrapStatus.Conditions[i], &condition) {
+			conditionChanged = true
+			clusterBootstrapStatus.Conditions[i] = condition
+		}
+	}
+	if !conditionExists {
+		conditionChanged = true
+		clusterBootstrapStatus.Conditions = append(clusterBootstrapStatus.Conditions, condition)
+	}
+
+	return conditionChanged, nil
+}
+
+// watchPackageInstalls sets a remote watch on the provided cluster on the Kind resource
+func (r *PackageInstallStatusReconciler) watchPackageInstalls(ctx context.Context, cluster *clusterapiv1beta1.Cluster, log logr.Logger) error {
```

Comment:
question: Trying to understand what are the tradeoffs we want to pass in a logger instead of using `r.Log` ?

Reply:
In the first line of ```Reconcile```, we set ```log := r.Log.WithValues(constants.ClusterNamespaceLogKey, req.Namespace, constants.ClusterNameLogKey, req.Name)``` and pass that along. The benefit of that is that every logs (info or error) will hold cluster namespaced information and can help in debugging. Otherwise, we should have that using ```WithValues()``` to every single log message and would be a pain and if forgotten to add, the log information would not be as much useful for debugging purposes.
```logr.Logger``` has an interface(2 words) and an int, so it would be 3 words to pass along. It has been passed along in almost all functions in ClusterBootstrap controller as well.

Reply:
Filed https://github.com/vmware-tanzu/tanzu-framework/issues/1831 to enhance current usages in the addons code